### PR TITLE
SLING-12087: Add metadata to the PackageMessage

### PR DESCRIPTION
**Pull request by jose-correia**

### CODE REVIEW

**Review by tmaret — COMMENTED** [review]
Could we add a test that ensure a PackageMessage produced before that change can be parsed with empty metadata ?

**Review by tmaret — APPROVED** [review]

**Review by tmaret — APPROVED** [review]

**Review by tmaret — CHANGES_REQUESTED** [review]
The PR breaks the previous constructor (without the metadata).

The PR should provide the old constructor.

File: src/main/java/org/apache/sling/distribution/journal/messages/PackageMessage.java [hunk 1]
```diff
@@ -41,20 +39,42 @@ public class PackageMessage {
...
+
     @Builder.Default
     List<String> paths = new ArrayList<>();
-    
+
     @Builder.Default
     List<String> deepPaths = new ArrayList<>();
-    
+
+    Map<String, String> metadata;
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/messages/PackageMessage.java]: Could we initialise this with an empty map in order to avoid having to check for null ?

**Review by tmaret — CHANGES_REQUESTED** [review]

**Review by tmaret — APPROVED** [review]
